apps/src/weblab2/CDOIDE/cdo-ide-context/cdo-ide-context-project-reducer.ts

```diff
@@ -0,0 +1,273 @@
+import {ProjectType, ReducerAction, FileId, FolderId} from '@cdoide/types';
```

Comment:
(can be a follow up) redux code should be written with redux toolkit

Reply:
Sure, but this isn't redux code. It's a context. ;-) 

I'd _strenuously_ argue against moving the state management into redux, because keeping it inside of the separate compartment inside of the ide hierarchy keeps it as a tidy black box. yeah, yeah, we can compartmentalize it here inside a separate redux store, but that pattern is frowned upon within the same repo.

Regardless, we can have the follow up discussions about where to place it as a follow up, for sure.

Reply:
gotcha, makes sense